lib/src/rules/weak_mode_unnecessary_null_checks.dart

```diff
@@ -0,0 +1,132 @@
+// Copyright (c) 2020, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:analyzer/dart/analysis/features.dart';
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/token.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+import 'package:analyzer/dart/element/element.dart';
+
+import '../analyzer.dart';
+
+const _desc = r'Unnecessary null check for non-nullable value.';
+
+const _details = r'''
+
+Unnecessary null check for non-nullable value in a opted-out library.
+
+**BAD:**
+```
+nonNullable?.property;
+if (nonNullable != null) {}
+```
+
+**GOOD:**
+```
+nonNullable.property;
+```
+
+''';
+
+class WeakModeUnnecessaryNullChecks extends LintRule implements NodeLintRule {
+  WeakModeUnnecessaryNullChecks()
+      : super(
+            name: 'weak_mode_unnecessary_null_checks',
+            description: _desc,
+            details: _details,
+            group: Group.style);
+
+  @override
+  void registerNodeProcessors(
+      NodeLintRegistry registry, LinterContext context) {
+    if (context.isEnabled(Feature.non_nullable)) {
+      return;
+    }
+
+    final visitor = _Visitor(this, context);
+    registry.addPropertyAccess(this, visitor);
+    registry.addMethodInvocation(this, visitor);
+    registry.addPostfixExpression(this, visitor);
+    registry.addBinaryExpression(this, visitor);
+  }
+}
+
+class _Visitor extends SimpleAstVisitor<void> {
+  _Visitor(this.rule, this.context);
+
+  final LintRule rule;
+  final LinterContext context;
+
+  @override
+  void visitPropertyAccess(PropertyAccess node) {
+    if (node.operator.type == TokenType.QUESTION_PERIOD &&
+        isNonNullable(node.target)) {
+      rule.reportLintForToken(node.operator);
+    }
+  }
+
+  @override
+  void visitMethodInvocation(MethodInvocation node) {
+    if (node.operator?.type == TokenType.QUESTION_PERIOD &&
+        isNonNullable(node.target)) {
+      rule.reportLintForToken(node.operator);
+    }
+  }
+
+  @override
+  void visitPostfixExpression(PostfixExpression node) {
+    if (node.operator.type == TokenType.QUESTION &&
```

Comment:
I don't believe that the operator of a `PostfixExpression` can ever be a question mark.